worker/pricetable.go

```diff
@@ -0,0 +1,156 @@
+package worker
+
+import (
+	"context"
+	"errors"
+	"math"
+	"sync"
+	"time"
+
+	"go.sia.tech/renterd/internal/consensus"
+	"go.sia.tech/renterd/rhp/v3"
+	rhpv3 "go.sia.tech/renterd/rhp/v3"
+	"go.sia.tech/siad/types"
+)
+
+type priceTables struct {
+	mu          sync.Mutex
+	priceTables map[consensus.PublicKey]*priceTable
+}
+
+type priceTable struct {
+	pt     *rhp.HostPriceTable
+	hk     consensus.PublicKey
+	expiry time.Time
+
+	w *worker
```

Comment:
does it make sense to use an interface here? something like `Transporter` or something. I had to look to see why you needed a worker for. I think you should keep the worker reference on the `priceTables` struct by the way, there's no need for it to be on every `priceTable` right?